contracts/escrow/LemonadeEscrowV1.sol

```diff
@@ -0,0 +1,328 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/finance/PaymentSplitter.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import "./ILemonadeEscrow.sol";
+import "./ILemonadeEscrowFactory.sol";
+
+bytes32 constant ESCROW_DELEGATE_ROLE = keccak256("ESCROW_DELEGATE_ROLE");
+
+contract LemonadeEscrowV1 is
+    ILemonadeEscrow,
+    PaymentSplitter,
+    AccessControlEnumerable
+{
+    using ECDSA for bytes;
+    using ECDSA for bytes32;
+
+    bool public closed;
+    uint16 public hostRefundPercent;
+
+    RefundPolicy[] _refundPolicies;
+    mapping(uint256 => bool) _paymentCancelled;
+    mapping(uint256 => Deposit[]) _deposits;
+    ILemonadeEscrowFactory _factory;
+
+    constructor(
+        address owner,
+        address[] memory delegates,
+        address[] memory payees,
+        uint256[] memory shares,
+        uint16 refundPercent,
+        RefundPolicy[] memory refundPolicies,
+        address factory
+    ) PaymentSplitter(payees, shares) {
+        if (refundPercent > 100) {
+            revert InvalidHostRefundPercent();
+        }
+
+        hostRefundPercent = refundPercent;
+
+        //-- check valid refundPolicies
+        uint256 refundPoliciesLength = refundPolicies.length;
+
+        if (refundPoliciesLength == 1) {
+            RefundPolicy memory policy = refundPolicies[0];
+
+            _assertValidRefundPercent(policy);
+            _refundPolicies.push(policy);
+        } else if (refundPoliciesLength > 1) {
+            RefundPolicy memory current;
+            RefundPolicy memory next;
+
+            next = refundPolicies[0];
+
+            _assertValidRefundPercent(next);
+            _refundPolicies.push(next);
+
+            for (uint256 i = 1; i < refundPoliciesLength; ) {
+                current = next;
+                next = refundPolicies[i];
+
+                if (
+                    current.timestamp >= next.timestamp ||
+                    current.percent <= next.percent
+                ) {
+                    revert InvalidRefundPolicies();
+                }
+
+                _assertValidRefundPercent(next);
+                _refundPolicies.push(next);
+
+                unchecked {
+                    ++i;
+                }
+            }
+        }
+
+        _grantRole(DEFAULT_ADMIN_ROLE, owner);
+        _grantRole(ESCROW_DELEGATE_ROLE, owner);
+
+        uint256 delegatesLength = delegates.length;
+        for (uint256 i; i < delegatesLength; ) {
+            _grantRole(ESCROW_DELEGATE_ROLE, delegates[i]);
+            unchecked {
+                ++i;
+            }
+        }
+
+        _factory = ILemonadeEscrowFactory(factory);
+    }
+
+    //- modifiers
+
+    modifier onlyDelegate() {
+        if (!hasRole(ESCROW_DELEGATE_ROLE, _msgSender())) {
+            revert AccessDenied();
+        }
+        _;
+    }
+
+    modifier escrowOpen() {
```

Comment:
Modifiers should go last in the contract

Reply:
this contradicts with the recommendation from Solidity

Inside each contract, library or interface, use the following order:

Type declarations

State variables

Events

Errors

Modifiers

Functions